source/difference.js

```diff
@@ -26,8 +26,15 @@ var difference = _curry2(function difference(first, second) {
   var out = [];
   var idx = 0;
   var firstLen = first.length;
+  var secondLen = second.length;
+  var toFilterOut = new _Set();
+
+  for (var i = 0; i < secondLen; i += 1) {
+    toFilterOut.add(second[i]);
+  }
+
   while (idx < firstLen) {
-    if (!_includes(first[idx], second) && !_includes(first[idx], out)) {
+    if (toFilterOut.add(first[idx])) {
```

Comment:
Any reason not to use the has function?

Reply:
That`s because I need to keep track of every value that was inserted into `out` before (for uniqueness), so `.add` does this job and conveniently let me know if the value is new or repeated.